test/language/arguments-object/mapped/mapped-arguments-nonconfigurable-descriptors-4.js

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2017 Caio Lima. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+info: Mapped arguments property descriptor change with non-configurable property
+description: >
+    Mapping keep working when property is set to non-configurable and its
+    value is changed using arguments[i] where 'i' is the argument index.
+flags: [noStrict]
+---*/
+
+function argumentsAndSetByIndex(a) {
+  Object.defineProperty(arguments, "0", {configurable: false});
+
+  arguments[0] = 2;
+  let propertyDescriptor = Object.getOwnPropertyDescriptor(arguments, '0');
+  assert.sameValue(propertyDescriptor.value, 2);
```

Comment:
I was in doubt to add it, since it is already being tested on test/language/arguments-object/mapped/mapped-arguments-nonconfigurable-4.js

I'm also in doubt if these cases should be added as a new JS test cases or if we can just add descriptor asserts on current tests.